aws/resource_aws_dynamodb_table.go

```diff
@@ -816,15 +816,21 @@ func flattenAwsDynamoDbTableResource(d *schema.ResourceData, meta interface{}, t
 	if err != nil {
 		return err
 	}
-	timeToLive := []interface{}{}
-	attribute := map[string]*string{
-		"name": timeToLiveOutput.TimeToLiveDescription.AttributeName,
-		"type": timeToLiveOutput.TimeToLiveDescription.TimeToLiveStatus,
-	}
-	timeToLive = append(timeToLive, attribute)
-	d.Set("timeToLive", timeToLive)
 
-	log.Printf("[DEBUG] Loaded TimeToLive data for DynamoDB table '%s'", d.Id())
+	if timeToLiveOutput.TimeToLiveDescription.AttributeName != nil {
```

Comment:
Any reason we're checking `timeToLiveOutput.TimeToLiveDescription.AttributeName` instead of `timeToLiveOutput.TimeToLiveDescription` here?

I guess we can assume `AttributeName` is not `nil` if the parent struct is not `nil`?

Reply:
We cannot actually. 😄 It causes a panic on `AttributeName` dereference in the basic acceptance testing without the additional `nil` check. I will update it to check both the parent struct and `AttributeName`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2537d6f]

goroutine 375 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.flattenAwsDynamoDbTableResource(0xc42019b650, 0x3217440, 0xc420532480, 0xc4207e8160, 0x0, 0x5168700)
	/Users/bflad/go/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_dynamodb_table.go:823 +0x1a1f
```